tensorflow/compiler/mlir/python/mlir.cc

```diff
@@ -133,6 +148,87 @@ std::string ImportFunction(const std::string &functiondef_proto,
                                  status);
 }
 
+std::string ImportGraphDef(const std::string& proto,
+                           const std::string& pass_pipeline,
+                           bool show_debug_info, TF_Status* status) {
+  GraphDebugInfo debug_info;
+  GraphImportConfig specs;
+  return ImportGraphDefImpl(proto, pass_pipeline, show_debug_info, debug_info,
+                            specs, status);
+}
+
+std::string ImportGraphDef(const std::string& proto,
+                           const std::string& pass_pipeline,
+                           bool show_debug_info, absl::string_view input_names,
+                           absl::string_view input_data_types,
+                           absl::string_view input_data_shapes,
+                           absl::string_view output_names, TF_Status* status) {
+  std::vector<string> node_names = absl::StrSplit(input_names, ',');
+  std::vector<string> node_dtypes = absl::StrSplit(input_data_types, ',');
+  std::vector<string> node_shapes_str = absl::StrSplit(input_data_shapes, ':');
+  std::vector<std::vector<int>> node_shapes;
+
+  std::vector<int> dims;
+  for (unsigned i = 0, e = node_shapes_str.size(); i < e; i++) {
+    dims.clear();
+    for (const auto& dim_str : absl::StrSplit(node_shapes_str[i], ',')) {
+      // Treats empty input shape as scalar.
+      if (dim_str.empty()) continue;
```

Comment:
Shouldn't this happen before the loop? I'm not sure which case this is catching otherwise?

```
   if (!node_shapes_str[i].empty()) {
    for (const auto& dim_str : absl::StrSplit(node_shapes_str[i], ',')) {
```

Reply:
@joker-eph This is only there to populate the input_shapes. Could you explain your point?

Reply:
I'm asking what kind of specification would have an empty `dim_str` here?

Reply:
I understand your point, makes sense not to evaluate the loop in that case.